pkg/secretless/plugin/connector/README.md

```diff
@@ -0,0 +1,267 @@
+through the `connector.Resources` interface, which is passed to your plugin's
+constructor.  You will need to retain a reference to `connector.Resources`, via
+a closure, inside the `Connector` function returned by your constructor.
+Here's the `connector.Resources` interface:
+
+```go
+package connector
+
+type Resources interface {
+  Config()          []byte
+  Logger()          secretless.Logger
+}
+```
+
+`Logger()` provides a basic logger you can use for debugging and informational
+logging.  `Config()` provide you resources specified in your `secretless.yml`
+file.
+
+Let's break down each method:
+
+- `Config()` - Some connectors require additional, connector-specific
+  configuration.  Anything specified in your connector's `config` section will
+  be passed back via this method as a raw `[]byte`.  Your code is responsible
+  for casting those bytes back into a meaningful `struct` that your code can
+  work with.
+- `Logger()` - Returns an object similar to the standard library's
+  `log.Logger`.  This lets you log events to stdout and stderr.  It respects
+  Secretless's command line `debug` flag, so that calling `Debugf` or `Infof`
+  will do nothing unless you started Secretless in debug mode.  See below for
+  details.
+
+#### `secretless.Logger` Interface
+
+Your code should never use the `fmt` or `log` packages, or write directly to
+stdout or stderr.  Instead, call `Logger()` on your `connector.Resources` to get
+a `secretless.Logger` with the following interface:
+
+```go
+type Logger interface {
+  Debugf(format string, v ...interface{})
+  Debug(msg string)
+  Debugln(msg string)
+
+  Infof(format string, v ...interface{})
+  Info(msg string)
+  Infoln(msg string)
+
+  Warnf(format string, v ...interface{})
+  Warn(msg string)
+  Warnln(msg string)
+
+  Errorf(format string, v ...interface{})
+  Error(msg string)
+  Errorln(msg string)
+
+  Fatalf(format string, v ...interface{})
+  Fatal(msg string)
+  Fatalln(msg string)
+}
+```
+
+The 3 `Debug` methods and the 3 `Info` methods do nothing unless you started
+Secretless with the `Debug` command line flag set to true.  If you did start
+Secretless in debug mode, they write to stdout.
+
+The `Warn`, `Error`, and `Fatal` methods all write to stderr, regardless of
+the current `Debug` mode.  
+
+All the `Fatal` methods call `os.Exit(1)` after printing their message.
+
+The `Logger` automatically prepends the name of the currently running service
+to all messages.  That is, the service name specified in your `secretless.yml`.
+For example, if your `secretless.yml` looks like:
+
+```
+version: "v2"
+services:
+  sample-service:
+    protocol: pg
+    listenOn: unix:///sock/.s.PGSQL.5432
+    ...
+```
+
+then the `Logger` will prepend `sample-service` to all messages.
+
+## Examples
+
+To see an example external connector, please take a look at our [test plugin](../../../../test/plugin).
+You can also look at our internal plugins (which also implement this interface):
```

Comment:
You can also look at our internal plugins that also implement this interface. 

Not sure why some of the sentence was in parens?